_posts/2021-01-19-release-1.12.1.md

```diff
@@ -0,0 +1,231 @@
+---
+layout: post
+title:  "Apache Flink 1.12.1 Released"
+date:   2021-01-19 00:00:00
+categories: news
+authors:
+- xtsong:
+  name: "Xintong Song"
+  twitter: "xtsong"
+---
+
+The Apache Flink community released the first bugfix version of the Apache Flink 1.12 series.
+
+This release includes 79 fixes and minor improvements for Flink 1.12.0. The list below includes a detailed list of all fixes and improvements.
+
+We highly recommend all users to upgrade to Flink 1.12.1.
+
+<div class="alert alert-danger small" markdown="1">
+<b>Attention:</b>
+Using UnalignedCheckpoints in Flink 1.12.0 combined with two/multiple inputs tasks or with union inputs for single input tasks can result in corrupted state.
+
+This can happen if a new checkpoint is triggered before recovery is fully completed. For state to be corrupted a task with two or more input gates must receive a checkpoint barrier exactly at the same time this tasks finishes recovering spilled in-flight data. In such case this new checkpoint can succeed, with corrupted/missing in-flight data, which will result in various deserialisation/corrupted data stream errors when someone attempts to recover from such corrupted checkpoint.
+</div>
+
+Updated Maven dependencies:
+
+```xml
+<dependency>
+  <groupId>org.apache.flink</groupId>
+  <artifactId>flink-java</artifactId>
+  <version>1.12.1</version>
+</dependency>
+<dependency>
+  <groupId>org.apache.flink</groupId>
+  <artifactId>flink-streaming-java_2.11</artifactId>
+  <version>1.12.1</version>
+</dependency>
+<dependency>
+  <groupId>org.apache.flink</groupId>
+  <artifactId>flink-clients_2.11</artifactId>
+  <version>1.12.1</version>
+</dependency>
+```
+
+You can find the binaries on the updated [Downloads page]({{ site.baseurl }}/downloads.html).
+
+Notice:
+<ul>
+<li>The DockerHub <a href='https://hub.docker.com/_/flink'>official images</a> for Apache Flink 1.12.1 are temporary missing. As an alternative, the images are currently available on the Flink PMC managed <a href='https://hub.docker.com/r/apache/flink'>registry</a>. This is now also the default for the Native Kubernetes deployments. The Flink PMC will keep working with the DockerHub team to deliver the official images.</li>
+<li>The source and python 3.8 linux wheel packages for Apache Flink 1.12.1 are temporary missing on <a href='https://pypi.org/project/apache-flink'>PyPI</a>, due to the project space limit. The request for increasing the space limit is currently under the PyPI review process. During this time, attempts to use the missing packages will fall back to the 1.12.0 packages.</li>
```

Comment:
What about removing `During this time, attempts to use the missing packages will fall back to the 1.12.0 packages.` as I guess it may cause confusion for users as this only holds when all the following conditions hold:
- users install pyflink without specifying the version, e.g. "pip install apache-flink" 
- Python version is 3.8
- The OS is linux

Besides, what about adding `During this period, you can <a href='https://ci.apache.org/projects/flink/flink-docs-release-1.12/flinkDev/building.html#build-pyflink'>build the package manually</a> if needed.` to guide users what to do during this time period?